app/models/audit/display_item.rb

```diff
@@ -0,0 +1,129 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+# Generic wrapper around a paper trail version to assist with display logic
+class Audit::DisplayItem
+  attr_reader :version, :username, :impersonating, :error, :changes, :entity_name, :entity_display_name
+  delegate :created_at, :event, :item_type, :item_id, to: :version
+
+  def initialize(version, users_by_id, excluded_fields = [])
+    raise ArgumentError, "Expected GrPaperTrail::Version or GrdaWarehouse::Version, got #{version.class}" unless version.is_a?(GrPaperTrail::Version) || version.is_a?(GrdaWarehouse::Version)
+
+    @version = version
+    @excluded_fields = excluded_fields
+
+    true_username = compute_username(users_by_id, version.clean_true_user_id)
+    username = compute_username(users_by_id, version.clean_user_id)
+    @username = true_username || username || version.whodunnit.presence
```

Comment:
Probably want to replace blank strings too.
```suggestion
    @username = true_username.presence || username.presence || version.whodunnit
```

Reply:
Actually, it looks like `compute_username` handles the presence, so you can probably ignore this.